cmd/testgrid-config-generator/main.go

```diff
@@ -170,21 +163,27 @@ func main() {
 			return fmt.Errorf("could not unmarshal release controller config at %s: %v", path, err)
 		}
 
-		var product, version string
-		if releaseConfig.Publish.Mirror.ImageStreamRef.Name != "" {
-			product = "okd"
-			version = releaseConfig.Publish.Mirror.ImageStreamRef.Name
-		} else if releaseConfig.Publish.Tag.TagRef.Name != "" {
+		var product string
+		switch {
+		case strings.HasSuffix(releaseConfig.Name, ".nightly"):
 			product = "ocp"
-			version = releaseConfig.Publish.Tag.TagRef.Name
-		} else {
-			logrus.Infof("could not determine publish destination for config at %v", path)
+		case strings.HasSuffix(releaseConfig.Name, ".ci"):
+			product = "okd"
+		default:
+			logrus.Infof("release is not recognized: %s", releaseConfig.Name)
 			return nil
 		}
+		m := reVersion.FindStringSubmatch(releaseConfig.Name)
```

Comment:
Do we have to? Ugh.

Reply:
Yeah, the publish tags aren't guaranteed to match releases, names are